presentation/src/main/java/com/knocklock/presentation/util/Extensions.kt

```diff
@@ -0,0 +1,27 @@
+package com.knocklock.presentation.util
+
+import android.content.Context
+import android.content.Intent
+import android.graphics.Bitmap
+import android.graphics.ImageDecoder
+import android.net.Uri
+import android.os.Build
+import android.provider.MediaStore
+
+
+@Suppress("DEPRECATION", "NewApi")
+fun Uri.parseBitmap(context: Context): Bitmap {
+    context.grantUriPermission(context.packageName, this, Intent.FLAG_GRANT_READ_URI_PERMISSION)
+    val resolver = context.contentResolver.apply {
+        takePersistableUriPermission(this@parseBitmap, Intent.FLAG_GRANT_READ_URI_PERMISSION)
+    }
+    return when (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) { // 28
+        true -> {
+            val source = ImageDecoder.createSource(resolver, this)
+            ImageDecoder.decodeBitmap(source)
+        }
+        else -> {
+            MediaStore.Images.Media.getBitmap(resolver, this)
+        }
+    }
```

Comment:
```suggestion
    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) { 
            val source = ImageDecoder.createSource(resolver, this)
            ImageDecoder.decodeBitmap(source)
        } else {
            MediaStore.Images.Media.getBitmap(resolver, this)
        }
```

if문으로 해도 될 것 같아요! ㅋㅋㅋㅋ

Reply:
coil사용으로 삭제 되었습니다!